editor/header/tools/publish-button.js

```diff
@@ -51,7 +51,7 @@ function PublishButton( {
 	const className = classnames( 'editor-tools__publish-button', { 'is-saving': isSaving } );
 	const onClick = () => {
 		const doSave = isPublished ||
-			! process.env.NODE_ENV === 'production' ||
+			! process.env.NODE_ENV !== 'production' ||
```

Comment:
Should this not just be `process.env.NODE_ENV !== 'production'`? There are double negatives which make this confusing. The save should happen automatically (without `confirm`) if not in the production environment.